### ISSUE TITLE
Feature request: Decode nested messages

### ISSUE THREAD

**Issue by dvorst** [dannyhann/protobuf_decoder#7]
Feature request for the ability to decode nested protobuf messages.

Current workaround for this is:
```python
parsed_data = Parser().parse(hex_str)
for r in parsed_data.results:
    if isinstance(r.data, str) and '\x12' in r.data:
        r.data = Parser().parse(r.data.encode().hex())
```

**Comment by dannyhann** [comment]
Hello @dvorst 

I currently support nested protobuf in this library. Could you please let me know the version you've been using and the hex code you tested?

In the recent 0.2.0 version, I've altered the parsing priority. Now, it first tries to parse as a string and, in case of an error, falls back to nested parsing. There might be a potential issue with this change.

Also, you're using the \x12 byte. Could you please explain why you chose this? 

I appreciate your feedback.
Thank you.

**Comment by fuzzyrichie** [comment]
@dannyhann - any chance I could get the ability to push a named branch for a PR purpose? Or should I do this via a fork?

**Comment by dannyhann** [comment]
We are improve logic and released version `0.3.0` 🚀
so, Closing this issue 😀
